bot/community/community.py

```diff
@@ -173,6 +174,98 @@ async def delete_community_room(self, member: discord.Member, before: discord.Vo
             await txt_ch.delete(reason=reason)
             log.info("Empty Community Room [%s] has been automatically deleted.", before.channel.name)
 
+    @commands.Cog.listener(name='on_raw_reaction_add')
+    async def mark_as_highlight(self, payload: discord.RawReactionActionEvent):
+        """Event listener which triggers if a reaction has been added to a message.
+
+        If enough users react to a message with the specified highlight emoji, it will be reposted in the configured
+        highlights channel by SAM. This way even users who don't have access to specific channels are able to see
+        interesting content from somewhere on the server.
+        If recently a highlight message has already been posted for a specific message, the reaction counter inside its
+        embed will be modified.
+
+        Args:
+            payload (discord.RawReactionActionEvent): The payload for the triggered event.
+        """
+        if payload.emoji.name != const.EMOJI_HIGHLIGHT or payload.channel_id == int(const.CHANNEL_ID_HIGHLIGHTS) \
+                or self._db_connector.is_botonly(payload.channel_id):
+            return
+
+        guild = self.bot.get_guild(payload.guild_id)
+        message_channel = guild.get_channel(payload.channel_id)
+        message = await message_channel.fetch_message(payload.message_id)
+        reaction = next(x for x in message.reactions if x.emoji == const.EMOJI_HIGHLIGHT)
+
+        has_author_reacted = await reaction.users().get(id=message.author.id)
+        reaction_counter = reaction.count - 1 if has_author_reacted else reaction.count
+
+        highlight_channel = guild.get_channel(int(const.CHANNEL_ID_HIGHLIGHTS))
+        highlight_message = await _check_if_already_highlight(highlight_channel, message.id)
+
+        if highlight_message:
+            embed = highlight_message.embeds[0]
+            embed.set_field_at(0, name=f"{const.EMOJI_HIGHLIGHT} {reaction_counter}", value=const.ZERO_WIDTH_SPACE)
+            await highlight_message.edit(content=message.content, embed=embed)
+
+        elif reaction_counter == const.LIMIT_HIGHLIGHT:
```

Comment:
maybe do >= so if sth happens (eg. bot is offline) and the message has not been posted, it will be posted on the next star, if it has not been posted yet.